sentinel-demo/sentinel-demo-okhttp/src/main/java/com/alibaba/csp/sentinel/demo/okhttp/controller/OkHttpTestController.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 1999-2020 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.csp.sentinel.demo.okhttp.controller;
+
+import com.alibaba.csp.sentinel.adapter.okhttp.SentinelOkHttpInterceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+/**
+ * @author zhaoyuguang
+ */
+@RestController
+public class OkHttpTestController {
+
+    @Value("${server.port}")
+    private Integer port;
+
+    @RequestMapping("/okhttp/back")
+    public String back() {
+        return "Welcome Back!";
+    }
+
+    @RequestMapping("/okhttp/back/{id}")
+    public String back(@PathVariable String id) {
+        return "Welcome Back! " + id;
+    }
+
+    @RequestMapping("/okhttp/testcase/{id}")
+    public String testcase(@PathVariable String id) throws Exception {
+        return getRemoteString(id);
+    }
+
+    @RequestMapping("/okhttp/testcase")
+    public String testcase() throws Exception {
+        return getRemoteString(null);
+    }
```

Comment:
How about also adding demo for `post` method?

When I add a post controller for test:
```java
@PostMapping("/okhttp/post")
public String post(HttpServletRequest request) throws Exception {
    byte[] body = StreamUtils.copyToByteArray(request.getInputStream());
    System.out.println(new String(body));
    return "post";
}

@RequestMapping("/okhttp/testcase/post")
public String testcasePost() throws Exception {
    MediaType mediaType = MediaType.parse("application/json; charset=utf-8");
    String requestBody = "{\"name\":\"cdfive\"}";
    OkHttpClient client = new OkHttpClient.Builder()
            .addInterceptor(new SentinelOkHttpInterceptor())
            .build();
    Request request = new Request.Builder()
            .url("http://localhost:" + port + "/okhttp/post")
            .post(RequestBody.create(mediaType, requestBody))
            .build();
    Response response = client.newCall(request).execute();
    return response.body().string();
}
```
I find the resource name in dashboard are:
`http://localhost:8085/okhttp/back`
`http://localhost:8085/okhttp/post`
They have no `GET` or `POST`, and also no `okhttp:` prefix

This is due to the custom extrator in `OkHttpDemoApplication` class.
My point is whether the interface `OkHttpResourceExtractor#extract(Request request, Connection connection)`
granularity is too big? When user use their custom extrator, they can't be benefit from or reuse the logic of `DefaultOkHttpResourceExtractor`, which provide common `prefix` and `method` logic.

The `UrlCleaner` interface in `sentinel-web-servlet` module can be referred.
`UrlCleaner`: `String clean(String originUrl);`
and the `CommonFilter` append the http method prefix outside the `clean` method.

Now, the advantage of `DefaultOkHttpResourceExtractor` is that it allows users to define resources completely when use their custom extrator.
It may need take info consideration compromise in design and decide which is better.
How about your option? @sczyh30 @zhaoyuguang

Reply:
恩 prefix默认是空的，demo中 自定义提取器写的粗暴一点呢
```java
        SentinelOkHttpConfig.setExtractor(new OkHttpResourceExtractor() {
            @Override
            public String extract(Request request, Connection connection) {
                String url = request.url().toString();
                String regex = "/okhttp/back/";
                if (url.contains(regex)) {
                    url = url.substring(0, url.indexOf(regex) + regex.length()) + "{id}";
                }
                return url;
            }
        });
```
我还是建议给用户开放 竟可能多的参数，issue https://github.com/alibaba/Sentinel/issues/1287 反应的就像更高阶的定义资源

Reply:
开放更多的参数没问题，`prefix`:`method`:`url`，只是`prefix`:`method`看要不要固定，
比如：
`String extract(String target, Request request, Connection connection)` 
或者 `String extract(String url, Request request, Connection connection)`

个人觉得现在这样也不错，灵活性最大。只是正好想到这个参考下哈~

Reply:
target 和 url 是哪里来的呢？是okhttp-adapter默认提取出来的 然后提供给用户么？

Reply:
@zhaoyuguang 大概想到这2个思路：
1. 先用`prefix:method:url`处理好，架设处理后值为target，
然后通过接口`String extract(String target, Request request, Connection connection)`让用户进一步自定义处理，默认返回target。

2.先得到url，
然后通过接口`String extract(String url, Request request, Connection connection)`让用户自定义处理，默认返回url，
最后前面加上`prefix:method:`。

第1种是整体用前缀处理好，由用户进一步处理；
第2种是用户处理url后，再加前缀；
值得注意的是：两种里面的`prefix`如果不需要或者要自定义都可以通过`(SentinelOkHttpConfig#setPrefix)`设置；但method目前暂时不行。

上面2种就是有一定得规则约定，在此基础上自定义；
目前的方式是最灵活，如果用户自定义，则整个资源名都自定义。
@sczyh30 宿何空了也看看呢~

Reply:
这里叫 ResourceExtractor 的原因是不希望给用户“只能将 URL 作为资源名”的印象。建议 prefix 可以放到 extractor 外边，然后 `method:URL` 这种 就作为默认实现好了，用户有需要 就自定义。

Reply:
使用了 第二种 ResourceExtractor 方法是
String extract(String url, Request request, Connection connection);
url默认是`method:URL`

Reply:
传入的 URL 里面就不要带 `${method}:` 了，参数就按照 URL 来传入，只不过 default extractor 里面的实现带上这个 method 前缀；另外总的那个 prefix 带到 interceptor 里面来作为总体控制的，如果有配就 append 上来。

Reply:
> 另外总的那个 prefix 带到 interceptor 里面来作为总体控制的，如果有配就 append 上来。
是从interceptor 将 prefix 带到 extract 方法中么？
String extract(String prefix, String url, Request request, Connection connection);
然后default extractor 判断有就加在前面？

Reply:
Fixed 抽空 review下